tutorials/editor_basics/editor_events.md

```diff
@@ -2,172 +2,168 @@
+To acquire that handler, an instance of the [EditorActionManager](upsource:///platform/platform-api/src/com/intellij/openapi/editor/actionSystem/EditorActionManager.java) needs to be obtained first.
+Fortunately, the `EditorActionManager` class provides a static method to do this.
 
+To request the correct `EditorActionHandler` from `EditorActionManager`, consult the [IdeActions](upsource:///platform/platform-api/src/com/intellij/openapi/actionSystem/IdeActions.java) interface for the correct constant to pass into the `EditorActionManager.getActionHandler()` method.
+For cloning a caret below the primary caret the constant is `ACTION_EDITOR_CLONE_CARET_BELOW`.
+Based on that constant, the `EditorActionManager` returns an instance of [CloneCaretActionHandler](upsource:///platform/platform-impl/src/com/intellij/openapi/editor/actions/CloneCaretActionHandler.java), a subclass of `EditorActionHandler`.
 ```java
-public class EditorIllustration extends AnAction {
-    static {
-        final EditorActionManager actionManager = EditorActionManager.getInstance();
-        final TypedAction typedAction = actionManager.getTypedAction();
-        typedAction.setupHandler(new MyTypedHandler());
-    }
+public class EditorHandlerIllustration extends AnAction {
+  @Override
+  public void actionPerformed(@NotNull AnActionEvent e) {
+    final Editor editor = e.getRequiredData(CommonDataKeys.EDITOR);
+    EditorActionManager actionManager = EditorActionManager.getInstance();
+    EditorActionHandler actionHandler = actionManager.getActionHandler(IdeActions.ACTION_EDITOR_CLONE_CARET_BELOW);
+   }
+
+   @Override
+   public void update(@NotNull final AnActionEvent anActionEvent) { /*...*/ }
 }
 ```
 
-After compiling and running the code snippet above typing in the editor will be handled with inserting an extra string at the 0 position.
-
-## 3.2. Working with EditorActionHandler
-
-Class
-[EditorActionHandler.java](upsource:///platform/platform-api/src/com/intellij/openapi/editor/actionSystem/EditorActionHandler.java)
-is used for actions activated by keystrokes in the editor.
-The steps below show how to access *EditorActionManager* and pass it the actions to be executed.
-In this example we will use *EditorActionHandler* to insert one extra caret below the current caret if applicable.
-
-### 3.2.1. Pre-requirements
-
-Create an action:
-
+### Using an EditorActionHandler to Clone the Caret
+To actually clone the caret requires only calling the `EditorActionHandler.execute()` method and passing in the appropriate context.
 ```java
 public class EditorHandlerIllustration extends AnAction {
-    @Override                                        
-    public void actionPerformed(@NotNull AnActionEvent anActionEvent) {
-    }
-    @Override
-    public void update(@NotNull final AnActionEvent anActionEvent) {
-    }
+  @Override
+  public void actionPerformed(@NotNull AnActionEvent e) {
```

Comment:
could we avoid repeating the nearly exact code sample from previous paragraph here?

Reply:
I'm open to suggestions about how to split the topics of acquiring an EditorActionHandler and using it to clone the caret without repeating some code.

I think tutorials by nature are going to be more repetitive than other forms of documentation.

Reply:
Combined sections on editor_events.md and coordinate_system.md to repeat code less on each page.